text/0000-unify-katello-ui.md

```diff
@@ -0,0 +1,78 @@
+- Feature Name: Unify Katello UI
+- Start Date: 2016-08-26
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The goal of this feature is to combine [Katello](github.com/katello/katello) and [Bastion](github.com/katello/bastion)
+into a single UI project.
+
+# Motivation
+[motivation]: #motivation
+
+Currently when working on the UI in Katello a developer has to oftentimes make changes in both the Katello and Bastion
+projects in order to complete a feature.  Once Katello begins to make more use of the components from
+[angular-patternfly](https://github.com/patternfly/angular-patternfly) this problem will only become worse as there will
+be three potential UI repositories to alter in order to complete new features.
+
+The expected outcome of this feature is to reduce the complexity of working on UI code in Katello by combining projects.
+
+# Detailed design
+[design]: #detailed-design
+
+## Option 1:  Move Bastion code back into Katello
+
+Eliminate Bastion entirely and move everything in Bastion back into Katello.
+
+### Pros
+- One less repository to manage
+- Would no longer require the release of a Bastion gem to update UI code in Katello
+- UI changes would be easier because they would only be made in one repository/project
+
+### Cons
+- Decreases separation of concerns, the UI would be bundled with the API code (although you could count this as a pro
+since it would bring us into consistency with [Foreman](https://github.com/theforeman/foreman))
+
+## Option 2:  Move Katello UI code into Bastion
+
+Keep Bastion and move all AngularJS based Katello UI code into Bastion
+
+### Pros
+- UI code and API code is separated
+- Potential to remove rails from the UI code
+
+### Cons
+- Still have two repositories/projects to manage
+- Would have to release the UI code to see changes in the main project
```

Comment:
This is true, it would become similar to how the CLI is handled today with both relying heavily on the API. Sometimes I think that separation of concern is good for development, and forces developers to think about each aspect and how it would be used. As there is that tendency when they are all together to make one big PR that changes everything, making it harder to review from a length of PR perspective and from a muddying the waters perspective. I have to think about, all at once, what is the API change, how does that work for pure API users. What is the UI change, how does that look and feel, is the UI influencing the API design in a way thats bad for API users.